samples/client/petstore/python/petstore_api/model_utils.py

```diff
@@ -1747,7 +1771,8 @@ def get_allof_instances(self, model_args, constant_args):
     for allof_class in self._composed_schemas['allOf']:
 
         try:
-            allof_instance = allof_class(**model_args, **constant_args)
+            intersected_args = find_args_intersection(allof_class, model_args)
+            allof_instance = allof_class(**intersected_args, **constant_args)
```

Comment:
Every schema/class instance should ingest all arguments. Not the subset of intersected ones.
This is required by openapi. Why are you passing in the intersection only?

Reply:
This could be a misunderstanding of how a constructor for a composed object is supposed to work. If the composed schema can handle all the arguments (that could be handled by separate allOf instances), its possible that some of the allOf structures cannot handle them all. As per my pervious comment above, how can the TriangleInterface handle a ModelSimple class it knows nothing about (eg. the ShapeName example I gave)?

Reply:
Because ModelSimple classes types must be seen as the underlying base type str/int/float etc.
TriangleInterface should accept name/ShapeName as a str even though it is a ModelSimple. 
I left a comment about that on https://github.com/OpenAPITools/openapi-generator/issues/9684

Reply:
After reviewing the discussion in #9684 and the requirement that all properties must be applied to all (allOf) instances, it occured to me that the configuration property "discard_unknown_keys" will satisfy the requirement where certain types are not accepted by the allOf instances. I removed the find_args_intersection code, but kept the original test that now applies this configuration.

Reply:
Discard unknown keys may work for you. One other use case to be aware of is that and of these composed schemas A through E may have minproperties or maxProperties validations. Those validations require that a dict have a certain number of keys. Which again points us to each schema ingesting all payload params. In my opinionthe best way toffix this is to fix the get_simple class logic.

Reply:
would it suffice to leave this PR as is, and introduce that as another fix to resolve the general use case?

Reply:
This PR has done a great job identifying problems and working to solve them. For the reasons that I laid out above I don't want to merge the intersection code. 
How about separate prs to solve the discriminator issue and the ingesting ModelSimple issue with additionalProperties?
this could becomeoone ffo thosePRs.

Reply:
Thanks for the feedback. I removed the intersection code with the recent force push. The only thing this PR does is now solve the recursion issue (8579). It doesn't truly solve #9684, as per your comments. However, I have made no changes with regard to that, other than add a testcase that shows the behavior of using the discard to achieve this one use case.